### ISSUE TITLE
[Programmatic access - Make Code - Home]: Role attribute does not have an appropriate value for the element.

### ISSUE THREAD

**Issue by Keertisree** [microsoft/pxt-microbit#5475]
### **User Experience:**
Visually impaired users who depend on screen reader will not understand the exact purpose of the controls and will find it difficult to find role of the controls on the page.

**Note**: User credentials should NOT be included in the bug.

### **Repro Steps:**

1. Open URL: https://arcade.makecode.com/hardware/raspberry-pi and sign in with valid credentials.
2. Run Accessibility Insights tool. 
3. Navigate the page.
4. Verify whether elements with aria roles have all required attributes or not.

### **Actual Result:**
Role attribute does not have an appropriate value for the element.
Observation: Screen reader announcing as 'search button' but navigating with swift tab it is announcing as 'Menu search button to move items press down arrow' when focus moves to search button.

### **Expected Result:**
ARIA role form is not allowed for given element.

**Note**: The same issue exists throughout the application.

**Issue**: Ensures role attribute has an appropriate value for the element (aria-allowed-role - https://dequeuniversity.com/rules/axe/4.7/aria-allowed-role?application=msftAI)

**Target application**: Raspberry Pi - https://arcade.makecode.com/hardware/raspberry-pi

**Element path:** #tocsearch1

**Snippet**: <form id="tocsearch1" class="search" role="form" method="get" action="https://www.bing.com/search">

**How to fix:** 
Fix any of the following:
  ARIA role form is not allowed for given element.

Environment: Chrome version 121.0.0.0

====

This accessibility issue was found using Accessibility Insights for Web 2.41.0 (axe-core 4.7.2), a tool that helps find and fix accessibility issues. Get more information & download this tool at http://aka.ms/AccessibilityInsights.

### **MAS Reference:**
[MAS 4.1.2 – Name, Role, Value](https://aka.ms/MAS4.1.2)

### **ARIA Authoring Ref Link/Fluent Patterns Guidance /HITS Guidance:**

[ARIA role should be appropriate for the element | Axe Rules | Deque University | Deque Systems](https://dequeuniversity.com/rules/axe/4.7/aria-allowed-role?application=msftAI)

### **UIA/DOM details available:** 

Yes (Please refer to the attachment)

### **Reference Links:**
[Accessibility Insights!](https://accessibilityinsights.io/) - Identify accessibility bugs before check-in and make bug fixing faster and easier.
External Bug Process: If this bug belongs to an external team, mark it as resolved/done and assign it back to the tester with notes on where to file/route the bug. For more information, please use this [External Bug Process (sharepoint.com)](https://microsoft.sharepoint.com/teams/cea11y/SitePages/External-Bug-Process.aspx?xsdata=MDN8MDF8fDE1MThmM2JlZGFjMDRlZDhiZmViYTUxMDA2M2NiZWI3fDcyZjk4OGJmODZmMTQxYWY5MWFiMmQ3Y2QwMTFkYjQ3fDF8MHw2Mzc3NDEyNTM2MTEzNzIwNzR8R29vZHxWR1ZoYlhOVFpXTjFjbWwwZVZObGNuWnBZMlY4ZXlKV0lqb2lNQzR3TGpBd01EQWlMQ0pRSWpvaUlpd2lRVTRpT2lJaUxDSlhWQ0k2TVRKOQ%3D%3D&sdata=MnlpSXlVMnpGT3cvZ0xJNnFWUWZWTGt6dmo1VkJMZFpIZHBvUUduNDhyMD0%3D&ovuser=72f988bf-86f1-41af-91ab-2d7cd011db47%2cv-kgeetala%40microsoft.com). “HCL Staff should not log any third-party external bugs. Should be routed to EDAD team”.
Please reach out to [C&AI Teams channel](https://apc01.safelinks.protection.outlook.com/?url=https://teams.microsoft.com/l/team/19%253a3cd071821b694f11b054d878453cd75f%2540thread.skype/conversations?groupId%3D30e3f968-15dd-4da2-bd8e-85f27e4af2c3%26tenantId%3D72f988bf-86f1-41af-91ab-2d7cd011db47&data=02%7c01%7cAklavyaM%40hcl.com%7c24e89c6ac26f42cfe40c08d7667e49f3%7c189de737c93a4f5a8b686f4ca9941912%7c0%7c0%7c637090564906844600&sdata=22JjBMoA1ID/KnYqFa/OB%2BrkTaLh5YYsGBd/7Bm7u5w%3D&reserved=0) for any process-related queries.

<img width="1361" alt="MAS4 1 2-Role attribute does not have an appropriate value for the element" src="https://github.com/microsoft/pxt-microbit/assets/158543478/1cf554a5-949f-45d0-b540-05311cc7f477">

**Comment by Keertisree** [comment]
#Accessibility;#A11yMAS;#A11yAuto;#A11ySev2;#K4W;#HCL;#E2E-HCL-MakeCode-Jan2024;#Win11-chrome;#MAS4.1.2
